test/e2e/tests/test_trainingjob_debugger.py

```diff
@@ -0,0 +1,226 @@
+    return training_sm_desc["DebugRuleEvaluationStatuses"][0]["RuleEvaluationStatus"]
+
+def get_training_debugger_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    resource_status = resource["status"]["debugRuleEvaluationStatuses"][0]["ruleEvaluationStatus"]
+    assert resource_status is not None
+    return resource_status
+
+@service_marker
+@pytest.mark.canary
+class TestTrainingDebuggerJob:
+    def _wait_sagemaker_training_status(
+        self, 
+        training_job_name,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_training_sagemaker_status,
+            training_job_name,
+        )
+    
+    def _wait_resource_training_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status, 
+            wait_periods,
+            period_length, 
+            get_training_resource_status, 
+            reference
+    )
+
+    def _assert_training_status_in_sync(
+        self, training_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_training_status(
+                training_job_name, expected_status
+            ) 
+            == self._wait_resource_training_status(reference, expected_status)
+            == expected_status
+        )
+
+    def _wait_sagemaker_training_debugger_status(
+        self, 
+        training_job_name,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_training_debugger_sagemaker_status,
+            training_job_name,
+        )
+    
+    def _wait_resource_training_debugger_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status, 
+            wait_periods,
+            period_length,
+            get_training_debugger_resource_status, 
+            reference
+    )
+
+    def _assert_training_debugger_status_in_sync(
+        self, training_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_training_debugger_status(
+                training_job_name, expected_status
+            )
+            == self._wait_resource_training_debugger_status(reference, expected_status)
+            == expected_status
+        )
+    
+    def test_stopped(self, xgboost_training_job_debugger):
```

Comment:
do we need test stopped again, nothing different from TestTrainingJob::test_stopped right?

Reply:
fair point, but then if this is the only one running in the canary, we should keep it.